### PR TITLE
STORM-2215: validate blobs are present before submitting

### PR DESCRIPTION
**Pull request by revans2**

### CODE REVIEW

File: storm-core/src/jvm/org/apache/storm/StormSubmitter.java [hunk 1]
```diff
@@ -587,4 +592,10 @@ private static double getMaxExecutorMemoryUsageForTopo(StormTopology topology, M
         }
         return largestMemoryOperator;
     }
+
+    private static Set<String> getListOfKeysFromBlobStore(Map<String, Object> stormConf) {
+        NimbusBlobStore client = new NimbusBlobStore();
```
Comment (HeartSaVioR) [inline storm-core/src/jvm/org/apache/storm/StormSubmitter.java]: Need to guarantee calling client.shutdown().

File: storm-core/src/jvm/org/apache/storm/utils/Utils.java [hunk 2]
```diff
@@ -1338,6 +1339,26 @@ public static void handleUncaughtException(Throwable t) {
         }
     }
 
+    public static void validateTopologyBlobStoreMap(Map stormConf, Set<String> blobStoreKeys) throws InvalidTopologyException {
+        boolean containsAllBlobs = true;
```
Comment (HeartSaVioR) [inline storm-core/src/jvm/org/apache/storm/utils/Utils.java]: Minor: We're adding missing keys to Set so no need to have additional flag. `missingKeys.size() > 0` is same to `containsAllBlobs == false`.

**Comment by revans2** [comment]
@HeartSaVioR I rebased and addressed the review comments, but I want to merge in #1744 first and then I will update this one again.

**Comment by revans2** [comment]
#1744 is in and this is rebased

**Comment by HeartSaVioR** [comment]
+1 Thanks for the work.

**Pull request merged**